src/payload/rtr.rs

```diff
@@ -75,13 +69,44 @@ pub struct Ipv6Prefix {
 ///
 /// There is at least one more payload type – BGPSEC router keys – that is not
 /// currently supported by the crate.
-#[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+#[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+#[non_exhaustive]
 pub enum Payload {
-    /// An IPv4 route origin authorisation.
-    V4(Ipv4Prefix),
+    /// A route origin authorisation.
+    Origin(PrefixOrigin),
+
+    /// A BGPsec router key.
+    RouterKey(RouterKey),
```

Comment:
There are two PDUs for origins, IPv4 prefix and IPv6 prefix. I find the their names somewhat misleading, since they are really route origin authorizations for these prefixes. Since we now have a common type for both V4 and V6 prefixes, we only have one variant for both of them here and convert that variant into PDUs accordingly.